### ISSUE TITLE
Schema "text_key" issue, redundant property

### ISSUE THREAD

**Issue by pashva** [langchain-ai/langchain-weaviate#94]
The init function accepts text_key as input, yet it is not being passed to the _default_schema function. Consequently, a property named "text" is hardcoded within the property.

![image](https://github.com/langchain-ai/langchain-weaviate/assets/61597430/eddeee85-d76d-4d62-9c88-b7e02d767d34)

Can I create a PR fixing this?

**Comment by hsm207** [comment]
@pashva thanks for your interest to contribute!

I noticed this too and wanted to get feedback from the community. Here's how I see it:

There's a lot of things you can customise in a weaviate schema. How do we decide which one is worth exposing to langchain users? If we expose all possible settings, then were are just building another abstraction on top of the weaviate client.

If we expose only some settings, then I feel it is an inconvenience for the user since there's now different settings have different ways of tweaking.

A solution I've been considering is to tell users this is what you get with the default schema. If you want something different, then use the weaviate client to create your desired schema, and when you use langchain, remember to specify the index name and which property is the text_key.

What do you think of this?

**Comment by pashva** [comment]
Yes it makes sense completely but then asking text_key as an input kind of confuses me personally as i believe if i specify the text key then my end schema created should have nothing but that text_key.

**Comment by hsm207** [comment]
> asking text_key as an input kind of confuses me personally as i believe if i specify the text key then my end schema created should have nothing but that text_key

I've noted the confusion.

I believe this can be solved with proper documentation in the class' docstring and tutorials/user guides.

What do you think?

**Comment by dudanogueira** [comment]
Went ahead and implemented text_key properly while implementing `schema` here https://github.com/langchain-ai/langchain-weaviate/pull/254

Note, if you pass a custom schema, `text_key` and `use_multi_tenancy` will be ignore
